SimulationAgent/SimulationRunner.cs

```diff
@@ -263,6 +272,10 @@ public void Stop()
 
                 // Reset rateLimiting counters
                 this.rateLimiting.ResetCounters();
+
+                // Reset vm count on vmss
+                this.azureManagementAdapter.CreateOrUpdateVmssAutoscaleSettingsAsync(1)
+                    .Wait(TimeSpan.FromSeconds(VMSS_AUTOSCALE_SETTINGS_TIMEOUT_SECS));
```

Comment:
same here, let's not manage the inftrastructure here.  Also, blocking code with `.Wait` should be avoided, prefer using async/await instead